### PR TITLE
implement `guest add`

### LINKED ISSUE #53: As a new user, I want to add new guests with name and optional phone number, email, address, RSVP status and dietary requirements
... so that I can keep track of the people whom I invited to the wedding.

### PR DESCRIPTION
**Pull request by p-xp**
Resolves #53.

Currently, the GuestAddCommand throws an exception when it is executed.

Let's implement the logic of executing a GuestAddCommand.

This will allow users to add guests to WedLog

### CODE REVIEW

File: src/test/java/wedlog/address/logic/parser/GuestAddCommandParserTest.java [hunk 1]
```diff
@@ -1,10 +1,185 @@
...
+    }
+
+    @Test
+    public void parse_missingAddress_success() {
+        Guest expectedGuest = new GuestBuilder(BOB).withoutAddress().build();
+        assertParseSuccess(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB
+                + RSVP_DESC_BOB + DIETARY_DESC_BOB + TAG_DESC_FRIEND
+                + TAG_DESC_HUSBAND, new GuestAddCommand(expectedGuest));
+
+    }
```
Comment (samuelim01) [inline src/test/java/wedlog/address/logic/parser/GuestAddCommandParserTest.java:139]: Perhaps you could add similar tests `parse_missingRsvpStatus_success()` and `parse_missingDietaryRequirements_success()`?

File: src/test/java/wedlog/address/testutil/GuestBuilder.java [hunk 2]
```diff
@@ -139,6 +152,22 @@ public GuestBuilder withoutEmail() {
...
+    }
+
+    /**
+     * Sets the {@code Dietary Requirements} of the {@code Guest} that we are building.
+     */
+    public GuestBuilder withoutDietaryRequirements() {
+        this.dietaryRequirements = null;
+        return this;
+    }
+
```
Comment (samuelim01) [inline src/test/java/wedlog/address/testutil/GuestBuilder.java:170]: The guest constructor requires non null values for rsvp status and dietary requirements. Perhaps you could remove these methods?

```suggestion
```

**Review by samuelim01 — COMMENTED** [review]
Just a few nits, but LGTM!

**Comment by wasjoe1** [comment]
Very rigorous testing!👍 LGTM!

**Pull request merged**